File: capri-places-app-frontend/src/app/services/review.service.ts
import { ReviewDTO } from './../models/review-model';
import { Observable } from 'rxjs';
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';

@Injectable({
  providedIn: 'root'
})
export class ReviewService {

  readonly baseUrl: string = "http://localhost:8080/api/v1/reviews"

  constructor(
    private http: HttpClient
  ) { }


  getByPlaceId(placeId: number): Observable<any> {
    return this.http.get<any>(this.baseUrl + "/place/" + placeId);
  }

  getByUserId(userId: number): Observable<any> {
    return this.http.get<any>(this.baseUrl + "/user/" + userId);
  }

  getById(id: number): Observable<any> {
    return this.http.get<any>(this.baseUrl + "/" + id);
  }

  addReview(reviewDTO: ReviewDTO): Observable<any> {
    return this.http.post(`${this.baseUrl}/add`, reviewDTO);
  }

  deleteReview(id: number): Observable<any> {
    return this.http.delete(this.baseUrl + "/delete/" + id)
  }

  editReview(reviewDTO: ReviewDTO): Observable<any> {
    return this.http.put(`${this.baseUrl + "/edit"}`, reviewDTO)
  }

  checkIfPreviouslyReviewed(userId: number, placeId: number): Observable<any> {
    return this.http.get<any>(this.baseUrl + "/reviewed/" + userId + "/" + placeId);
  }
}

File: review-service/src/main/java/com/ironhack/reviewservice/service/ReviewService.java
package com.ironhack.reviewservice.service;

import com.ironhack.reviewservice.dao.Review;
import com.ironhack.reviewservice.dto.ReviewDTO;
import com.ironhack.reviewservice.dto.ReviewResponse;

import java.util.List;

public interface ReviewService {

    List<ReviewResponse> getByPlaceId(Long placeId);

    List<ReviewResponse> getByUserId(Long userId);

    Review getById(Long id);

    Review editReview(ReviewDTO reviewDTO);

    Review addReview(ReviewDTO reviewDTO);

    void deleteReview(Long id);

    boolean isPreviouslyReviewed(Long userId, Long placeId);
}
